driver-core/src/main/com/mongodb/internal/operation/QueryBatchCursor.java

```diff
@@ -318,6 +322,8 @@ private BsonDocument asGetMoreCommandDocument(final ServerCursor serverCursor) {
     private ServerCursor initFromQueryResult(final QueryResult<T> queryResult) {
         nextBatch = queryResult.getResults().isEmpty() ? null : queryResult.getResults();
         count += queryResult.getResults().size();
+        LOGGER.debug(format("Received batch of %d documents with cursorId %d from server %s", queryResult.getResults().size(),
```

Comment:
Note this will always format a string regardless of debug being enabled.

Reply:
See above.